### PR TITLE
Fix - GeoIP does not work on self-hosted with ingress.nginx

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Description

As reported in https://github.com/PostHog/posthog/issues/6334 and https://github.com/PostHog/charts-clickhouse/issues/132 the GeoIP plugin does not work on self-hosted deployments when `ingress.nginx` is enabled.

This PR fixes the issue by:
1. adding metadata annotations to the `ingress-nginx` service definition in order to enable LB proxy mode for AWS and DO deployments (we will validate Azure and GCP in another PR as their e2e tests are currently blocked)

3. enhancing the current integration test suite we have in CI. The major change is that we added a new test to verify if the source IP address of a random ingested event is part of a public range. This will hopefully catch any regression in the future.

## Type of change

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## How has this been tested?
CI should be passing

## Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ci/values/amazon_web_services.yaml [hunk 1]
```diff
@@ -5,3 +5,17 @@ ingress:
     enabled: true
 cert-manager:
   enabled: true
+
+#
+# For AWS ELB in L4 (TCP) mode, we need to enable some additional config
```
Comment (guidoiaquinti) [inline ci/values/amazon_web_services.yaml:10]: Those will eventually be moved as default when `ingress.nginx.enabled` is set to `true`

File: ci/values/digital_ocean.yaml [hunk 2]
```diff
@@ -5,3 +5,17 @@ ingress:
     enabled: true
 cert-manager:
   enabled: true
+
```
Comment (guidoiaquinti) [inline ci/values/digital_ocean.yaml:8]: Ditto

File: charts/posthog/tests/ingress.yaml [hunk 3]
```diff
@@ -32,3 +32,115 @@ tests:
           # Unfortunately we can't match the whole manifest as there are few
           # metadata values linked to the chart version
           path: spec
+
```
Comment (guidoiaquinti) [inline charts/posthog/tests/ingress.yaml:35]: This is not strictly part of this PR but it's just some additional unit tests for the ingress definition

**Review by tiina303 — APPROVED** [review]

**Comment by tiina303** [comment]
we should change https://github.com/PostHog/charts-clickhouse/blob/main/do_1_click_values.yaml too?

**Comment by guidoiaquinti** [comment]
Actually I think we can't make this working without making PostHog install via Helm a two step process. The problem and potential solution are well explained [here](https://github.com/compumike/hairpin-proxy#the-problem) but the tl;dr; is that we can't make the ACME challenge pass when using PROXY protocol due to a limitation in `cert-manager`.

File: ci/values/amazon_web_services.yaml (hunk 1 above)
Comment (guidoiaquinti) [reply ci/values/amazon_web_services.yaml:10]: I don't think there's an easy way to override those at runtime.

**Comment by guidoiaquinti** [comment]
This is now working 🎉 

I'm waiting DO support to understand if there's a way to get the LB IP address while using the `do-loadbalancer-hostname` annotation without using `doctl` or looking at the DO UI as without it we'll break the handy command we use in our docs:
```
export INGRESS_IP=$(kubectl get --namespace posthog ingress posthog -o jsonpath="{.status.loadBalancer.ingress[0].ip}") && \
echo "\n-----\n" && \
echo "Your PostHog installation is available at: http://$INGRESS_IP" && \
echo "\n-----\n"
```

In the meantime, feedback to the PR content are welcome!

File: .github/workflows/test-digitalocean-install.yaml [hunk 4]
```diff
@@ -13,7 +13,7 @@ on:
     paths:
       - ci/**
       - charts/**
-      - .github/workflows/test-digital-ocean-install.yaml
+      - .github/workflows/test-digitalocean-install.yaml
```
Comment (fuziontech) [inline .github/workflows/test-digitalocean-install.yaml:16]: 👍

**Review by fuziontech — APPROVED** [review]
This is fantastic - I'll be using a good bit of this for the automation interface as well 🚢

**Comment by guidoiaquinti** [comment]
I'll update the docs but it's unfortunately not possible to get the IP in DO

> Hello, 
> 
> Thank you for your patience. 
> 
> Our Kubernetes Experts provided an update on your question about getting the Load Balancer IP. Using the service.beta.kubernetes.io/do-loadbalancer-hostname the annotation specifies the hostname used for the Service status.Hostname instead of assigning status.IP
> 
> We do not have another way to get the Load Balancers IP address you would need to either use the cloud panel or doctl to get it. 
>
> If you have any further questions please let us know.
>
> Kind Regards, 
>
> Tyler J 
>
> DigitalOcean Customer Success
>

**Comment by guidoiaquinti** [comment]
Merging also if e2e GCP is failing. The issue seems to be unrelated.

**Pull request merged**